docs/REST/authentication.md

```diff
@@ -289,3 +289,62 @@ static class HttpContentExtensions
     }
 }
 ```
+### Java
+```java
+public CloseableHttpResponse execute(HttpUriRequest request, String credential, String secret)
+        throws IOException, URISyntaxException {
+    Map<String, String> authHeaders = sign(request, credential, secret);
+    authHeaders.forEach(request::setHeader);
+
+    return httpClient.execute(request);
+}
+
+private static Map<String, String> sign(HttpUriRequest request, String credential, String secret) 
+        throws URISyntaxException, IOException {
```

Comment:
I am wondering if the input is an object of `HttpUriRequest` (instead of a uri string), is the `URISyntaxException` possible?

Reply:
The `URISyntaxException` comes from lines 312 and 315 with the use of `URIBuilder`, where it thinks it is possible to throw an exception as it takes in the URI as a String and not as a URI object. As for `HttpUriRequest`, which in the case of the provider is actually a `HttpGet`, it does not validate that the Uri is valid when creating the `HttpGet`.